tests/integration/courses.test.js

```diff
@@ -1431,127 +1422,126 @@ describe('COURSES ROUTES - POST /courses/{_id}/trainee', () => {
+        expect(response.statusCode).toBe(200);
       });
 
-      ['coach', 'client_admin'].forEach((role) => {
-        it(`should return 403 as user is ${role} requesting on an other company`, async () => {
-          const payload = {
-            identity: { firstname: 'Coco', lastname: 'Bongo' },
-            local: { email: 'coco_bongo@alenvi.io' },
-            company: authCompany._id,
-            contact: { phone: '0689320234' },
-          };
-
-          authToken = await getToken(role);
-          const response = await app.inject({
-            method: 'POST',
-            url: `/courses/${intraCourseIdFromOtherCompany}/trainees`,
-            headers: { Cookie: `alenvi_token=${authToken}` },
-            payload,
-          });
-
-          expect(response.statusCode).toBe(403);
+      it('should return a 400 error if trainee exist, has no company, and missing company parameter', async () => {
+        const falsyPayload = { local: { email: traineeWithoutCompany.local.email } };
+        const response = await app.inject({
+          method: 'POST',
+          url: `/courses/${interb2bCourseIdFromAuthCompany}/trainees`,
+          payload: falsyPayload,
+          headers: { Cookie: `alenvi_token=${authToken}` },
         });
+
+        expect(response.statusCode).toBe(400);
       });
+    });
+  });
 
-      it('should return a 200 as user is course trainer', async () => {
+  describe('Other roles', () => {
+    const roles = [
+      { name: 'helper', expectedCode: 403 },
+      { name: 'planning_referent', expectedCode: 403 },
+      { name: 'coach', expectedCode: 200 },
+    ];
+    roles.forEach((role) => {
+      it(`should return ${role.expectedCode} as user is ${role.name}, requesting on his company`, async () => {
         const payload = {
           identity: { firstname: 'Coco', lastname: 'Bongo' },
           local: { email: 'coco_bongo@alenvi.io' },
           company: authCompany._id,
           contact: { phone: '0689320234' },
         };
 
-        authToken = await getTokenByCredentials(trainer.local);
+        authToken = await getToken(role.name);
         const response = await app.inject({
           method: 'POST',
           url: `/courses/${intraCourseIdFromAuthCompany}/trainees`,
           headers: { Cookie: `alenvi_token=${authToken}` },
           payload,
         });
 
-        expect(response.statusCode).toBe(200);
-        sinon.assert.calledWithExactly(sendinBlueTransporter);
+        expect(response.statusCode).toBe(role.expectedCode);
       });
     });
-  });
 
-  describe('inter_b2b vendor_role', () => {
-    beforeEach(async () => {
-      authToken = await getToken('vendor_admin');
-    });
-
-    it('should add user to inter b2b course', async () => {
-      const existingUserPayload = { local: { email: auxiliary.local.email }, company: authCompany._id };
+    it('should return 403 as user is trainer if not one of his courses', async () => {
```

Comment:
ce nom n'est pass tres clair non ? le if est etrange `should return 403 as user is trainer but not of this courses`